src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerPlugin.java

```diff
@@ -202,6 +203,8 @@ public PerformanceAnalyzerPlugin(final Settings settings, final java.nio.file.Pa
                 performanceAnalyzerController,configOverridesWrapper));
         scheduledMetricCollectorsExecutor.addScheduledMetricCollector(new MasterThrottlingMetricsCollector(
                 performanceAnalyzerController,configOverridesWrapper));
+        scheduledMetricCollectorsExecutor.addScheduledMetricCollector(new ShardIndexingPressureMetricsCollector(
```

Comment:
Let's add the check for `Class.forName(SHARD_INDEXING_PRESSURE_CLASS_NAME);` here. That way the collector never runs if the dependency is not available.